### PR TITLE
Revert this/super blacklist for function composition heuristic

### LINKED ISSUE #4925: Composed functions not put on own line anymore after 1.14.0 update
Getting these diffs a lot after the 1.14 update. Seems like the functional composition detection doesn't work anymore
```diff

         this.subscriptions.add(
             this.componentUpdates
-                .pipe(
-                    startWith(this.props),
-                    distinctUntilChanged(isEqual)
-                )
+                .pipe(startWith(this.props), distinctUntilChanged(isEqual))
                 .subscribe((props: Props) => {
                     const newParentPath = props.activePathIsDir ? props.activePath : dirname(props.activePath)
                     const queryParams = new URLSearchParams(this.props.history.location.search)
```
**Prettier 1.14.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMAWBLAzgOiwVwCMswAnDABxg2lwEMATBgCgB0oACL7nr9bHJAC2FaAhgBVCgzrws7Xot44KlOMyww6pGAHUM6Zv1wVSEClgCUAGg4Ns1KGEmwMAGwDCaOlADmcFmwAUQBHfDo3S0sFJVi8IhJyQnVTcywOAF4APg5gGNiCjnzC7gBfaM4lSxBrEHNqWmRQbTMAdwAFbQQsZBA6ADcIDAYakEJSOjAAazgYAGUKSYw-ZBhSfDha5aw4HXaJ3yE6ZAAzCJ3agCssAA8AIQnp2bm6ITgAGWW4U-PNkEXSDtSL1CHRCABPNzQUamZZ6YboZAADgADLVUjtdBMKL1THAgf1vrVSHAwhgSfs6Idjkgzm4LiAdkIMKt1n8sMtfG44ABFfAQeA-el-LSEfQMRFIABMtTWdHcnI8ECER16UDEo3wOwAKmCerTfqVSkA)

**Input:**
```jsx
this.subscriptions.add(
            this.componentUpdates
                .pipe(startWith(this.props), distinctUntilChanged(isEqual))
                .subscribe(props => {
                    
                })
        )
```

**Output:**
```jsx
this.subscriptions.add(
  this.componentUpdates
    .pipe(startWith(this.props), distinctUntilChanged(isEqual))
    .subscribe(props => {})
);
```

**Expected behavior:**
```jsx
this.subscriptions.add(
  this.componentUpdates
    .pipe(
      startWith(this.props),
      distinctUntilChanged(isEqual)
     )
    .subscribe(props => {})
);
```
as it was with 1.13

### PR DESCRIPTION
**Pull request by suchipi**
Fixes #4925

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: tests/functional_composition/functional_compose.js [hunk 1]
```diff
@@ -1,38 +1,44 @@
 compose(
-  sortBy(x => x), 
-  flatten, 
+  sortBy(x => x),
+  flatten,
```
Comment (suchipi) [inline tests/functional_composition/functional_compose.js]: Sorry, my editor did this

**Comment by j-f1** [comment]
How about keeping the blacklist for `this.compose`/`super.compose`, but still printing more complicated calls in the FP style?

**Comment by suchipi** [comment]
Would we keep `this.pipe`?

**Comment by j-f1** [comment]
I’d say we should have the same behavior for `pipe` as for `compose`.

**Comment by suchipi** [comment]
I think it's really important for RxJS users that `pipe` has FP style formatting.
Maybe we don't need it for compose unless it's being used in the curried call format that redux does. But I'm hesitant to change it.

**Comment by suchipi** [comment]
Good to merge?

**Pull request merged**